### DISCUSSION TITLE
Manual tool change

### DISCUSSION THREAD

**Discussion by kotlikm** [linuxcnc-probe-screen/probe-screen#72] [Q&A]
Was hoping I might be able to get some more direct help here. I have just configured PSNG on my linuxcnc machine, everything seems to work as should, until I run a program, I get the pop up to load the tool, and when I press OK, nothing happens. It shows the manual tool change macro is loaded, but it doesn't continue and run over to the tool setter to touch off the new tool. The weird thing is, if I rerun the intended program, it will go over to the tool setter as desired, and then run the program, until it gets to the next tool change and the stops running after I swap in a new tool and press ok. 

I am using GMOCCAPY if that makes a difference, there doesn't seem to be much if any documentation on setting up with this GUI vs AXIS.

**Comment by freisei** [comment]
It seems to be a Prolem with PSNG, which was under heavy development. I 
think its hanging anywhere in the macros. You should determinate the 
hanging ngc-file and macro-line and add your problem here: 
https://github.com/linuxcnc-probe-screen/probe-screen-ng/issues/71

greets freisei.

**Reply by kotlikm** [reply]
Do you have any recommendations on how to track that down? I am more mechanically inclined that software inclined. 

I would really like to get this to work, but it seems like the original V2 no longer is compatible with LCNC 2.8, and it seems like a ton was changed in the NG version that its much harder for me to try and trouble shoot.

**Comment by snelweg** [comment]
This project looks dead to me. It is a shame.

**Comment by alkabal** [comment]
Hello

For the moment I can’t work more to this project and it seem that kiall is away.

Asap i will restart to work to this project or to my fork that have some difference with this one.

What is yo problem with actual code ?

**Reply by kotlikm** [reply]
I dont know enough about how everything is configured with the python script, but there appear to be something that is stuck. Once the program asks for a new tool at a tool change, i load the new tool, press ok to confirm i have changed it, and then nothing happens, It stays frozen with the preview screen showing the tool change macro is loaded, but will not proceed to re-measure the tool and run the NC program.

**Comment by alkabal** [comment]
Why this topic is not viewable on GitHub ???

**Comment by alkabal** [comment]
Can you try to use my fork ?

**Reply by kotlikm** [reply]
I've got a backlog of parts I need to be running, but I will work on getting a config compiled with your fork in the near term and test it out when I have the opportunity next week hopefully.
